docs/package.md

```diff
@@ -0,0 +1,90 @@
+# The Coalton `(package ...)` Form
+
+In Coalton, the `package` form is used to define a package by
+specifying which dependencies to import, and which definitions to
+export.
+
+The form also establishes the current package: any definitions that
+follow `(package <name>)` will be contained within that package.
+
+## Defining a Package
+
+A package in Coalton is defined using the `package` form. The basic
+syntax is:
+
+``` coalton
+(package <package-name>
+  (import <other-package-name>
+          (<other-package-name2 as <other2>)
+          ...)
+  (import-from <other-package-name>
+          <symbol1>
+          <symbol2>
+          ...)
+  (export <symbol3>
+          <symbol4>
+          ...))
+```
+
+Here's an example:
+
+``` coalton
+(package my-package
+  (import coalton-library/list)
+  (export sum-even-numbers))
+```
+
+## Importing Dependencies
+
+You can import dependencies for your package using the `import` and
+`import-from` forms.
+
+The `import` form allows you to specify which packages your package
+depends on. You can also rename imported packages using the `as`
+keyword, as shown in the example below:
+
+``` coalton
+(package my-package
+  (import coalton-library/list as list)
+  (export sum-even-numbers))
+```
+
+The `import-from` form allows you to specify which symbols from an
+imported package should be made available within your package:
+
+``` coalton
+(package my-package
```

Comment:
```
(O package, my-package
  (export
    our 
    fearful 
    trip 
    is 
    done))

Reply:
can also allow parser to accept "exhort" for "export"